src/common.h

```diff
@@ -127,6 +129,29 @@ static string returnFeatureName(FeatureTypes type) {
 
 }
 
+
+class Cov{
+public:
+    //the vraiables used in the coverage calculation function
+    ofstream coverageOutput;
+    int binStart;
+	int binEnd;
+	int currentChr;
+	vector<unsigned int> sequencedBases;
+	map<unsigned int,string> position2contig;
+	vector<int> contigLength;
+	uint32_t contigsNumber;
+	
+	
```

Comment:
I think you should improve the design of this class. Write a proper constructor that initialise all the fields that at this point can become private (instead of public as they are now).
`coverageMain` is a bit strange as it wants as input variables that are already a part of the object.... which seems suspicious.